### PR TITLE
Fix and Upgrade Patreon Support

### PR DESCRIPTION
**Pull request by Investigamer**
# Description

## Patreon Fixes
- Handle paginated results
- Exclude Patreon's automatic 'Free' tier
- Exclude non-active patrons
- Ensure display patron tier reflects their highest backed tier
- Enforce sorting patrons by tier level (dollar amount)

## Minor Fixes
- Update Levenshtein in reqs and pre-commit-config for Python 3.12 support
- Exclude common virtual environment names from mypy tests

# Pre-Commit Notes

Right now pre-commit on my end is falsely asserting that the module `elasticsearch` has no attribute `Elasticsearch` (`MPCAutofill\cardpicker\search\search_functions.py` line **9**) so I had to force commit overriding the pre-commit hook, all other pre-commit checks passed.

# Checklist

- [x] I have installed `pre-commit` and run the hooks with `pre-commit run`.
- [x] I have updated any related tests for code I modified or added new tests where appropriate.
- [x] I have updated any relevant documentation or created new documentation where appropriate.

### CODE REVIEW

File: MPCAutofill/cardpicker/integrations/patreon.py [hunk 1]
```diff
@@ -81,28 +88,76 @@ def get_patrons(campaign_id: str, campaign_tiers: dict[str, SupporterTier]) -> O
...
+            # Skip non-active members
+            if mem["attributes"]["patron_status"] != "active_patron":
+                continue
+
+            # Skip members with no tiers
+            tiers = mem.get("relationships", {}).get("currently_entitled_tiers", {}).get("data", [])
+            if not tiers:
+                continue
+
+            # Figure out highest paid tier
```
Comment (ndepaola) [inline MPCAutofill/cardpicker/integrations/patreon.py:124]: it's slightly clearer if you do
```suggestion
            # Figure out highest paid tier for this patron (patrons can subscribe to multiple tiers)
```

**Review by ndepaola — APPROVED** [review]
looks good! just a couple of minor comments

btw you can get the tests on github actions passing if you add your google drive api key as a repo secret - iirc there are instructions in the wiki

File: MPCAutofill/cardpicker/integrations/patreon.py [hunk 2]
```diff
@@ -81,28 +88,76 @@ def get_patrons(campaign_id: str, campaign_tiers: dict[str, SupporterTier]) -> O
...
-        ]
+        results: list[Supporter] = []
+        for mem in res["data"]:
+
+            # Skip non-active members
+            if mem["attributes"]["patron_status"] != "active_patron":
+                continue
+
+            # Skip members with no tiers
+            tiers = mem.get("relationships", {}).get("currently_entitled_tiers", {}).get("data", [])
```
[the remark this answers is not in the document]
Comment (Investigamer) [reply MPCAutofill/cardpicker/integrations/patreon.py]: 👍

File: MPCAutofill/requirements.txt [hunk 3]
```diff
@@ -18,7 +18,7 @@ jsonschema~=4.20.0
 google-api-python-client~=2.95.0
 google-auth-httplib2~=0.1.0
 google-auth-oauthlib~=1.0.0
-Levenshtein~=0.21.1
+Levenshtein~=0.23.0
```
[the remark this answers is not in the document]
Comment (Investigamer) [reply MPCAutofill/requirements.txt:21]: Anytime! 🙌

**Review by ndepaola — APPROVED** [review]

**Pull request merged**